lib/python/qmk/keymap.py

```diff
@@ -98,3 +101,57 @@ def write(keyboard, keymap, layout, layers):
         keymap_fd.write(keymap_c)
 
     return keymap_file
+
+def find_keymaps(base_path, revision = "", community = False):
```

Comment:
QMK makes no distinction between base_path and revision. There is only the keyboard folder. Unless I'm missing some context this function signature should probably match that reality.

The docstring doesn't make it very clear what the difference between `find_keymaps()` and `list_keymaps()` is. It's possible we only need one of them, or that this one should more clearly be designated as a helper for `list_keymaps()`.

Reply:
I split the `keyboard_name` into 2 parts because it makes sense when filtering paths. `keyboards/base_path/revision` is checked and `revision` is always optional so instead of 2 path checks one is enough.

The naming here not the best, that's for sure. `find_keymaps()` is a helper for `list_keymaps()`. The only reason it exists because the path filtering for community layouts and under the keyboard's directory needs a *tiny* bit of difference. So `find_keymaps()` was introduced to reduce code duplication.
It should probably be a "private" function.